Makefile

```diff
@@ -137,11 +176,17 @@ clean:  cli-clean test-clean manager-clean deploy-clean
 .PHONY: docker-build
 # Build the docker image
 docker-build: generate fmt vet manifests
-	docker build . -t ${IMG}
+	docker build --build-arg git_version_arg=${GIT_VERSION_PATH}=${GIT_VERSION} \
+	--build-arg git_commit_arg=${GIT_COMMIT_PATH}=${GIT_COMMIT} \
+	--build-arg build_date_arg=${BUILD_DATE_PATH}=${BUILD_DATE} . -t ${DOCKER_IMG}:${DOCKER_TAG}
+	docker tag ${DOCKER_IMG}:${DOCKER_TAG} ${DOCKER_IMG}:${GIT_VERSION}
+	docker tag ${DOCKER_IMG}:${DOCKER_TAG} ${DOCKER_IMG}:latest
 	@echo "updating kustomize image patch file for manager resource"
-	sed -i'' -e 's@image: .*@image: '"${IMG}"'@' ./config/default/manager_image_patch.yaml
+	sed -i'' -e 's@image: .*@image: '"${DOCKER_IMG}:${GIT_VERSION}"'@' ./config/default/manager_image_patch.yaml
```

Comment:
I don't think this working right:

```bash
ll config/default/
total 24
drwxr-xr-x  7 tom  staff  224 Apr 24 05:47 ./
drwxr-xr-x  6 tom  staff  192 Apr 24 05:40 ../
-rw-r--r--  1 tom  staff  874 Apr 24 05:40 kustomization.yaml
drwxr-xr-x  3 tom  staff   96 Apr 24 05:40 manager/
-rw-r--r--  1 tom  staff  283 Apr 24 05:44 manager_image_patch.yaml
-rw-r--r--  1 tom  staff  283 Apr 24 05:47 manager_image_patch.yaml-e
drwxr-xr-x  4 tom  staff  128 Apr 24 05:40 rbac/
```

Not sure what the `manager_image_patch.yaml-e` file is, but its not referenced in the `kustomize.yaml` file.

After doing a `make docker-build` the only change to the `config` folder is:
```bash
diff --git a/config/default/manager_image_patch.yaml-e b/config/default/manager_image_patch.yaml-e
index bd97461a..c20df502 100644
--- a/config/default/manager_image_patch.yaml-e
+++ b/config/default/manager_image_patch.yaml-e
@@ -8,5 +8,5 @@ spec:
     spec:
       containers:
       # Change the value of image field below to your controller image URL
-      - image: kudobuilder/controller:latest
+      - image: kudobuilder/controller:v0.1.0
         name: manager
```

Reply:
> I don't think this working right:
> 
> ```shell
> ll config/default/
> total 24
> drwxr-xr-x  7 tom  staff  224 Apr 24 05:47 ./
> drwxr-xr-x  6 tom  staff  192 Apr 24 05:40 ../
> -rw-r--r--  1 tom  staff  874 Apr 24 05:40 kustomization.yaml
> drwxr-xr-x  3 tom  staff   96 Apr 24 05:40 manager/
> -rw-r--r--  1 tom  staff  283 Apr 24 05:44 manager_image_patch.yaml
> -rw-r--r--  1 tom  staff  283 Apr 24 05:47 manager_image_patch.yaml-e
> drwxr-xr-x  4 tom  staff  128 Apr 24 05:40 rbac/
> ```
> 
> Not sure what the `manager_image_patch.yaml-e` file is, but its not referenced in the `kustomize.yaml` file.
> 

That's right, its not used in the kustomize patch and will be ignored by it. However, we reference the original in https://github.com/kudobuilder/kudo/blob/211d7aa3c0cc66141afb88059857a78b10d946e6/config/default/kustomization.yaml#L26, the `-e` version serves as sort of predecessor to track changes to that patch. but as those changes are anyways tracked by GitHub I think it is a good question if we actually need it.

> After doing a `make docker-build` the only change to the `config` folder is:
> 
> ```shell
> diff --git a/config/default/manager_image_patch.yaml-e b/config/default/manager_image_patch.yaml-e
> index bd97461a..c20df502 100644
> --- a/config/default/manager_image_patch.yaml-e
> +++ b/config/default/manager_image_patch.yaml-e
> @@ -8,5 +8,5 @@ spec:
>      spec:
>        containers:
>        # Change the value of image field below to your controller image URL
> -      - image: kudobuilder/controller:latest
> +      - image: kudobuilder/controller:v0.1.0
>          name: manager
> ```

I think this was introduced in https://github.com/kudobuilder/kudo/commit/705589c885684d0728509370e964dfc6bc198538 and borrowed eg from https://github.com/kubernetes-sigs/cluster-api/blob/master/Makefile#L131. It essentially modifies the patch file and stores a old copy under the `-e` version, so when you run the `docker-build` and the version changed you should expect the new version in the original patch file and the previous version in the `-e` file.

If you see no changes then maybe there was no version change at all or you maybe run into something similar as described here: https://github.com/kubernetes-sigs/cluster-api/issues/661 Then we might need a drop in wrapper as work around  in `hack/sed.sh` ( similar to https://github.com/kubernetes-sigs/cluster-api/blob/master/hack/sed.sh ) as oppose to the default macOS sed.